meson.build

```diff
@@ -0,0 +1,37 @@
+# This file is adapted from https://github.com/scipy/scipy/blob/main/meson.build
+project(
+    'pandas',
+    'c', 'cpp', 'cython',
+    version: '1.6.0.dev0',
```

Comment:
How does this version interact with some of the other work you've done to get versioneer to work?

Reply:
versioneer determines the __version__ attribute. The version here is currently only used be meson-python(incorrectly) to determine the wheel filename(we are working on getting meson-python to use the __version__ tag as well).

In the future, this'll probably only be a fallback version, if everything fails to detect the pandas version properly.

Reply:
I am still a bit wary of using `meson-python` coming from the angle that it is relatively immature. Sorry if I lost the answer to this question but what is the advantage of using meson-python versus just creating a simple setuptools hook to invoke meson as needed? The latter wouldn't require any changes to versioneer or other setuptools-based tooling @rgommers

Reply:
From https://github.com/lithomas1/pandas/pull/19#issuecomment-1259182255:

_@WillAyd it is possible, but I think that opens a whole new can of worms for no real reason. For example, you build the sdist with setuptools while you're building the wheel with a mix Meson and setuptools, leading to possible inconsistencies. And you'll be making cross-compiling worse rather than better._

_If you want a gradual transition, it'd be better to merge this PR first, but still default to `setuptools` in `pyproject.toml`. That allows devs to start using Meson, without all contributors and users who are pip-installing from GitHub getting it straight away. And then it's a patch of a few lines to switch that default later._

In addition, I think it's optimistic to think that you can write a hook like that which won't have bugs or surprising behavior. 

> I am still a bit wary of using `meson-python` coming from the angle that it is relatively immature.

That's understandable. I'm sure there's a couple of bugs left to discover. That said, it's now been in SciPy `main` since last December, and in the SciPy 1.9.x release series. So hopefully the most glaring defects have been found by now. And `meson` and `meson-python` are quite actively maintained, so any painful bugs can hopefully be remedied quickly.

You also have a number of months till the next release, right? Can always just try it, and in case you're not happy right before the next release, it's literally a one-line patch to revert back to setuptools for one more release cycle.